src/linked_list.py

```diff
@@ -16,25 +22,30 @@ def push(self, contents):
 
     def pop(self):
         """Remove and return the current head node."""
+        if not self.head_node:
+            raise AttributeError("List is empty")
         old_head_node = self.head_node
         self.head_node = self.head_node.next_node
         self.length -= 1
-        return old_head_node
+        return old_head_node.contents
 
     def size(self):
         """Return the current size of this linked list."""
         return self.length
 
     def search(self, search_value):
         """Return the node with the searched contents if found."""
-        if search_value == self.head_node.contents:
-            return self.head_node
-        current_node = self.head_node
-        while current_node.contents != search_value:
-            if current_node.next_node is None:
-                return None
-            current_node = current_node.next_node
-        return current_node
+        if self.length:
+            if search_value == self.head_node.contents:
+                return self.head_node
+            current_node = self.head_node
+            while current_node.contents != search_value:
+                if current_node.next_node is None:
+                    return None
+                current_node = current_node.next_node
+            return current_node
+        else:
+            return None
 
     def remove(self, remove_node):
         """Remove a node from linked list."""
```

Comment:
`remove` method is still throwing errors when trying to remove from an empty LinkedList.